### PR TITLE
[FLINK-16955] Bump Zookeeper 3.4.X to 3.4.14

### PR DESCRIPTION
**Pull request by piyushnarang**
Follow up from https://github.com/apache/flink/pull/11938 to commit in the right project. 
Picking up the updated zk dependency allows us to get around an issue in the `StaticHostProvider` in the current zk where if 1 of n configured zk hosts is unreachable the Flink job manager is not able to start up.

### CODE REVIEW

**Comment by piyushnarang** [comment]
cc @zentol - I tried using this version on Flink and I hit the issues that were captured in https://issues.apache.org/jira/browse/FLINK-11259
We need to make a minor tweak to the `SecureTestEnvironment.prepare(..)` to get around this - https://github.com/piyushnarang/flink/commit/fd6cc0311d0ef20606ba4566a54315a186889304

File: flink-shaded-zookeeper-parent/flink-shaded-zookeeper-34/pom.xml [hunk 1]
```diff
@@ -128,4 +128,4 @@ under the License.
             </dependency>
         </dependencies>
     </dependencyManagement>
-</project>
```
[the remark this answers is not in the document]
Comment (piyushnarang) [reply flink-shaded-zookeeper-parent/flink-shaded-zookeeper-34/pom.xml:131]: yes, will do. Missed this when I put it up

**Comment by piyushnarang** [comment]
Yeah, let me double check if things continue to work after the excludes.

**Comment by piyushnarang** [comment]
@zentol added the updates you requested. I did some basic sanity checking / testing after excluding the spotbugs and jsr305 and it seems to work ok. Do you know if there's a way to trigger the Flink full CI run for flink-shaded changes?

**Comment by zentol** [comment]
@piyushnarang There is unfortunately no _easy_ way of doing that.
I'd suggest to modify `tools/ci/maven-utils.sh` in the Flink project to checkout your branch, install it, and modifying the version property of Flink.

**Comment by piyushnarang** [comment]
Ok, I've already done that degree of sanity checking and ran a few of the tests on my machine locally.

**Pull request merged**